pkg/controller/vsphere/reconciler.go

```diff
@@ -217,12 +232,17 @@ func (r *Reconciler) delete() error {
 		}
 		if moTask != nil {
 			if taskIsFinished, err := taskIsFinished(moTask); err != nil {
-				metrics.RegisterFailedInstanceDelete(&metrics.MachineLabels{
-					Name:      r.machine.Name,
-					Namespace: r.machine.Namespace,
-					Reason:    "Task finished with error",
-				})
-				return err
+				// Check if latest task is not a task for vm cloning
+				if taskIsFinished && moTask.Info.DescriptionId != cloneVmTaskDescriptionId {
```

Comment:
Yep, it would be the same object in taskRef which was handled within `create` procedure. So, its possible to grab desription out of it.

https://github.com/openshift/machine-api-operator/pull/854/files#diff-21a982202fdcdf4beffc4d00659798b54717940bf07f685006dbcb99d9dbf0ecR119